frontend/packages/console-app/src/redux/reducers/quick-start-reducer.ts

```diff
@@ -1,31 +1,143 @@
+import { Map, fromJS } from 'immutable';
 import { QuickStartSidebarActions, Actions } from '../actions/quick-start-actions';
 import { RootState } from '@console/internal/redux';
+import {
+  AllQuickStartStates,
+  QuickStartState,
+  QuickStartStatus,
+  QuickStartTaskStatus,
+} from '../../components/quick-starts/utils/quick-start-types';
 
 export const quickStartReducerName = 'quickStart';
 
-type State = {
-  isExpanded: boolean;
-  activeQuickStartID: string;
-};
+export const QUICK_START_REDUX_STATE_LOCAL_STORAGE_KEY = 'bridge/quick-start-redux-state';
+
+type State = Map<string, any>;
```

Comment:
```suggestion
type State = Map<string, { activeQuickStartId: string }>;
```
I think we know more than `any`. It's a constructed state that we are saving... can we elaborate the type? I added the only one I saw in this file.

Reply:
Actually I cannot define it's type like `Map<string, { activeQuickStartId: string }>`. `activeQuickStartId` is the key to the `Map` itself and then it's value is of type `string` as well. But the second key there would be `allQuickStartStatuses` and it's value would be again of type `Map<string, any>`. Even the main `UI` redux reducer state follows the same type. i couldn't come up with a smart way to define all the type safe params here.

Reply:
Open to suggestion though.

Reply:
Fair enough.